ops/testing.py

```diff
@@ -1509,27 +1509,75 @@ def _get_backend_calls(self, reset: bool = True) -> List[Tuple[Any, ...]]:
-                unit) that will own the secret.
+            owner: The name of the remote application or a specific remote unit,
+                that will own the secret.
             content: A key-value mapping containing the payload of the secret,
                 for example :code:`{"password": "foo123"}`.
 
         Return:
-            The ID of the newly-secret added.
+            The ID of the newly-added secret.
         """
         owner_name = _get_app_or_unit_name(owner)
         model.Secret._validate_content(content)
         return self._backend._secret_add(content, owner_name)
 
+    def add_model_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
+        """An alias for :meth:`add_charm_secret` for backward compatibility."""
+        warnings.warn("add_model_secret is deprecated, use add_charm_secret instead.",
+                      category=DeprecationWarning)
+        return self.add_charm_secret(owner, content)
+
+    def add_user_secret(self, content: Dict[str, str]) -> str:
+        """Add a secret owned by the user, simulating the ``juju add-secret`` command.
+
+        Args:
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+
+        Return:
+            The ID of the newly-added secret.
+
+        Example usage (the parameter ``harness`` in the test function is
+        a pytest fixture that does setup/teardown, see :class:`Harness`)::
+
+            # config.yaml
+            options:
+              mysec:
+                type: secret
+                description: "tell me your secrets"
+
+            # charm.py
+            class MyVMCharm(ops.CharmBase):
+                def __init__(self, framework: ops.Framework):
+                    super().__init__(framework)
+                    framework.observe(self.on.config_changed, self._on_config_changed)
+
+                def _on_config_changed(self, event: ops.StartEvent):
+                    mysec = self.config.get('mysec')
+                    if mysec:
+                        sec = self.model.get_secret(id=mysec, label="mysec")
+                        self.config_from_secret = sec.get_content()
+
+            # test_charm.py
+            def test_config_changed(harness):
+                secret_content = {'password': 'foo'}
+                secret_id = harness.add_user_secret(secret_content)
+                harness.grant_secret(secret_id, 'webapp')
+                harness.begin()
+                harness.update_config({'mysec': secret_id})
+                secret = harness.model.get_secret(id=secret_id).get_content()
+                assert harness.charm.config_from_secret == secret.get_content()
+
+        """
+        model.Secret._validate_content(content)
+        # Although it's named a user-owned secret in juju, technically, the owner is the
+        # Model, so the secret's owner is set to :attr:`Model.uuid <ops.Model.uuid>`.
```

Comment:
I don't think it makes sense to use ReST for comments.

```suggestion
        # Model, so the secret's owner is set to `Model.uuid`.
```